src/main/scala/ScalaModulePlugin.scala

```diff
@@ -147,6 +147,17 @@ object ScalaModulePlugin extends AutoPlugin {
 
   lazy val scalaModuleSettingsJVM: Seq[Setting[_]] = scalaModuleOsgiSettings
 
+  lazy val scalaModuleSettingsNative: Seq[Settings[_]] = Seq(
+    scalaVersion := "2.11.11",
+    skip in compile := System.getProperty("java.version").startsWith("1.6"),
+    test := {},
+    libraryDependencies := {
+      if (!scalaVersion.value.startsWith("2.11"))
+        libraryDependencies.value.filterNot(_.organization == "org.scala-native")
+      else libraryDependencies.value
+    }
```

Comment:
It's as simple as that because aggregation and `+` have always been broken in sbt, except for the most simple setups. (`++` is fine). You don't need platform cross-compilation to be in a scenario where part of the projects in your build will not compile against all Scala versions. For example, if you write an sbt plugin, and extract its core logic in a separate project, you can cross-compile the core logic for all versions of Scala, but the sbt plugin can only build on 2.10.x (and now 2.12 with sbt 1.x).

Therefore, I argue that relying on aggregation and/or `+` was broken first, and adding cross-projects into the mix did not make the situation any worse. In fact, even though I've never used sbt-doge, I suspect it would solve the present situation with cross-projects just as well as the sbt plugin/core logic situation.